app/src/main/java/com/example/parstagram/fragments/DetailFragment.java

```diff
@@ -0,0 +1,278 @@
+                                }
+                                object.saveInBackground();
+                                ivLike.setImageResource(R.drawable.ufi_heart);
+                                int numQueryLikes = 0;
+                                try {
+                                    numQueryLikes = queryLikes(post);
+                                } catch (ParseException ex) {
+                                    ex.printStackTrace();
+                                }
+                                if (numQueryLikes == 1) {
+                                    String text = "Liked by 1 person";
+                                    tvLikes.setText(text);
+                                } else {
+                                    String text = "Liked by " + numQueryLikes + " people";
+                                    tvLikes.setText(text);
+                                }
+                            }
+                        });
+                    } else {
+                        Like like = new Like();
+                        like.setUser(ParseUser.getCurrentUser());
+                        like.setPost(post);
+                        like.saveInBackground(new SaveCallback() {
+                            @Override
+                            public void done(ParseException e) {
+                                Post post = (Post) getArguments().getSerializable("post");
+                                if (e != null) {
+                                    Log.e(TAG, "Error liking post", e);
+                                    return;
+                                }
+                                ivLike.setImageResource(R.drawable.ufi_heart_active);
+                                DrawableCompat.setTint(
+                                        DrawableCompat.wrap(ivLike.getDrawable()),
+                                        ContextCompat.getColor(getContext(), R.color.red)
+                                );
+                                int numQueryLikes = 0;
+                                try {
+                                    numQueryLikes = queryLikes(post);
+                                } catch (ParseException ex) {
+                                    ex.printStackTrace();
+                                }
+                                if (numQueryLikes == 1) {
+                                    String text = "Liked by 1 person";
+                                    tvLikes.setText(text);
+                                } else {
+                                    String text = "Liked by " + numQueryLikes + " people";
+                                    tvLikes.setText(text);
+                                }
+                            }
+                        });
+                    }
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
```

Comment:
You're doing a little too much with this function. I'd recommend refactor each action to a separate function & use more helper ones